app/services/immunity_details_creation_service.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class ImmunityDetailsCreationService
+  def initialize(planning_application:)
+    @planning_application = planning_application
+  end
+
+  def call
+    transaction do
+      immunity_detail = ImmunityDetail.new(planning_application: @planning_application)
+      immunity_detail.end_date = application_end_date
+      immunity_detail.save
+    end
```

Comment:
Not a blocker but maybe we could raise an error here so we can monitor if something goes wrong.